app/controllers/posts_controller.rb

```diff
@@ -49,9 +51,7 @@ def update
   end
 
   def destroy
-    unless current_user == @post.user
-      redirect_to root_path, alert: 'You are not authorized to delete this post.'
-    end
+    redirect_to root_path, alert: 'You are not authorized to delete this post.' unless current_user == @post.user
```

Comment:
If you will be restrict the access to some actions or views, you should send a status code like:

`redirect_to root_path, alert: 'You are not authorized to delete this post.', status: :unauthorized unless current_user == @post.user`

By this way, the server is telling the client that this action is unauthorized